cast/src/lib.rs

```diff
@@ -1656,6 +1661,63 @@ impl SimpleCast {
     pub fn disassemble(bytecode: &str) -> Result<String> {
         format_operations(disassemble_str(bytecode)?)
     }
+
+    /// Gets the selector for a given function signature
+    /// Optimizes if the `optimize` parameter is set to a number of leading zeroes
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use cast::SimpleCast as Cast;
+    ///
+    /// fn main() -> eyre::Result<()> {
+    ///     assert_eq!(Cast::get_selector("foo(address,uint256)", None)?.0, String::from("0xbd0d639f"));
+    ///
+    ///     Ok(())
+    /// }
+    /// ```    
+    pub fn get_selector(signature: &str, optimize: Option<usize>) -> Result<(String, String)> {
+        let optimize = optimize.unwrap_or(0);
+        if optimize > 4 {
+            eyre::bail!("Number of leading zeroes must not be greater than 4");
+        }
+        if optimize == 0 {
+            let selector = HumanReadableParser::parse_function(signature)?.short_signature();
+            return Ok((format!("0x{}", hex::encode(selector)), String::from(signature)))
+        }
+        let Some((name, params)) = signature.split_once('(') else {
+            eyre::bail!("Invalid signature");
+        };
+
+        let num_threads = num_cpus::get();
+        let found = AtomicBool::new(false);
```

Comment:
I believe we don't need this anymore if we use `find_map_any`? Since `rayon` will handle shutdown for us: <https://docs.rs/rayon/latest/rayon/iter/trait.ParallelIterator.html#method.find_map_any>

Reply:
not sure why but looks like it doesn't shutdown for a while. significantly slower vs using `AtomicBool`

Reply:
Ah I see, it doesn't cancel currently running "tasks" so all have to return before it selects one. Then it's fine

Reply:
yes since we use long worker threads we need to shut them down manually